include/alpaka/core/Common.hpp

```diff
@@ -137,7 +137,7 @@
     || BOOST_LANG_HIP)
 #    define ALPAKA_STATIC_ACC_MEM_GLOBAL                                                                              \
         template<typename TAcc>                                                                                       \
-        inline __device__
+        __device__
```

Comment:
This breaks the semantic of global variables when relocatable device compilation is used, because it makes them effectively `static`, so separate files refer to separate instances of the global variable.

Now, the macro does say `static` in the name.

Should we make it `static` also in intent, and implement a separate one for global variables with external linkage ?  Or just fix this one to behave as we expect, despite then name ?

Reply:
After some tests with and without RDC (relocatable device code) with both `nvcc` and `clang`, the most robust approach seems to be
  - for code compiled with the default flags, use `static`;
  - for code compiled with RDC (`nvcc -rdc`, `clang++ -fgpu-rdc`), use `inline`.

One can check for both with
```c++
#if defined __CUDACC_RDC__ || defined __CLANG_RDC__
#define ALPAKA_STATIC_ACC_MEM_GLOBAL template<typename TAcc> __device__ inline
#else
#define ALPAKA_STATIC_ACC_MEM_GLOBAL template<typename TAcc> __device__ static
#endif
```

This also works for `hipcc`, which uses the `-fgpu-rdc` flag and the `__CLANG_RDC__` macro.

@AuroraPerego could you rebase this PR and try to use the macro above ?

@SimeonEhrig do we have CI tests with and without RDC ?

Reply:
Done, let's see how the tests go (I'm afraid the CI doesn't test the `-rdc` flag, but I've run the tests for CUDA 12.4 with gcc 13.1 using `alpaka_RELOCATABLE_DEVICE_CODE=ON` and they passed).
Did you manage to compile CUDA code with `clang` or is this expected to fail if tested?

Reply:
There is a custom job, who tests it: https://github.com/alpaka-group/alpaka/blob/690da679bd209212541a95bb4f8e8be7c486bb5a/script/gitlabci/job_cuda.yml#L4

It uses CUDA 12.0 and gcc 12. You can update it and/or add more tests to test nvcc + gcc combinations, where you know, that they behaves different. I think CUDA 12.0 + gcc 12 is the "default" case and CUDA 12.4  + gcc 13 a "special" case, If don't understand the problem wrong.

But in general, we should also add the `alpaka_RELOCATABLE_DEVICE_CODE` to the generator.

Reply:
> Did you manage to compile CUDA code with `clang` or is this expected to fail if tested?

I got it to work, it seems `clang` does not support some multi-file cases that work with `nvcc`.
Probably not an issue with alpaka, because all code usually ends up in header files.